pkg/agent/route/route_windows.go

```diff
@@ -213,109 +220,116 @@ func (c *Client) DeleteRoutes(podCIDR *net.IPNet) error {
 // Service traffic from host network to OVS via antrea-gw0.
 func (c *Client) addVirtualServiceIPRoute(isIPv6 bool) error {
 	linkIndex := c.nodeConfig.GatewayConfig.LinkIndex
+	svcIP := config.VirtualServiceIPv4
 
-	// This route is for 2 purposes:
-	// - If for each ClusterIP Service, a route is installed to direct traffic to antrea-gw0, there will be too many
-	//   neighbor cache entries. While with one virtual IP to antrea-gw0 and ClusterIP Services to the virtual IP,
-	//   there will be only one neighbor cache entry.
-	// - For ClusterIP Service requests from host, reply traffic needs a route entry to route packet to VirtualServiceIPv4
-	//   via antrea-gw0. If the NextHop of a route is antrea-gw0, then set it as 0.0.0.0. As a result, on-link/0.0.0.0
-	//   is in PersistentStore.
-	// - For NodePort Service, it is the same.
-	vRoute := &util.Route{
-		LinkIndex:         linkIndex,
-		DestinationSubnet: virtualServiceIPv4Net,
-		GatewayAddress:    net.IPv4zero,
-		RouteMetric:       util.MetricHigh,
-	}
-	if err := util.ReplaceNetRoute(vRoute); err != nil {
-		return err
+	neigh := generateNeigh(svcIP, linkIndex)
+	if err := util.ReplaceNetNeighbor(neigh); err != nil {
+		return fmt.Errorf("failed to add new IP neighbour for %s: %w", svcIP, err)
 	}
-	klog.InfoS("Added virtual Service IP route", "route", vRoute)
+	klog.InfoS("Added virtual Service IP neighbor", "neighbor", neigh)
 
-	// Service replies will be sent to OVS bridge via openflow.GlobalVirtualMAC. This NetNeighbor is for
-	// creating a neighbor cache entry to config.VirtualServiceIPv4.
-	vNeighbor := &util.Neighbor{
-		LinkIndex:        linkIndex,
-		IPAddress:        config.VirtualServiceIPv4,
-		LinkLayerAddress: openflow.GlobalVirtualMAC,
-		State:            "Permanent",
-	}
-	if err := util.ReplaceNetNeighbor(vNeighbor); err != nil {
-		return err
-	}
-	klog.InfoS("Added virtual Service IP neighbor", "neighbor", vNeighbor)
-
-	if err := c.addServiceRoute(config.VirtualNodePortDNATIPv4); err != nil {
-		return err
-	}
-	// For NodePort Service, a new NetNat for NetNatStaticMapping is needed.
-	err := util.NewNetNat(antreaNatNodePort, virtualNodePortDNATIPv4Net)
-	if err != nil {
-		return err
+	route := generateRoute(virtualServiceIPv4Net, net.IPv4zero, linkIndex, util.MetricHigh)
+	if err := util.ReplaceNetRoute(route); err != nil {
+		return fmt.Errorf("failed to install route for virtual Service IP %s: %w", svcIP.String(), err)
 	}
+	c.serviceRoutes.Store(svcIP.String(), route)
+	klog.InfoS("Added virtual Service IP route", "route", route)
 
 	return nil
 }
 
-// TODO: Follow the code style in Linux that maintains one Service CIDR.
-func (c *Client) addServiceRoute(svcIP net.IP) error {
-	obj, found := c.hostRoutes.Load(svcIP.String())
-	svcIPNet := util.NewIPNet(svcIP)
+func (c *Client) AddClusterIPRoute(svcIP net.IP) error {
```

Comment:
Just confirm: no local cache for clusterIP route entry?

Reply:
Sorry, local cache? you mean sync.Map or something else?